### ISSUE TITLE
Global menu patch fails to merge

### ISSUE THREAD

**Issue by darthoctopus** [BrowserWorks/waterfox#350]
Previously I've been able to build Waterfox with Ubuntu's global menu patch (see e.g. [this](bazaar.launchpad.net/~mozillateam/firefox/firefox-trunk.head/view/head:/debian/patches/unity-menubar.patch)) matching the appropriate version of firefox. As of waterfox 56 this is no longer possible (the object files compile successfully but I get a linker error when linking libxul.so).

Ancillarily, would it be possible to merge in the global menu by default?

**Comment by criztovyl** [comment]
fixed link
https://bazaar.launchpad.net/%7Emozillateam/firefox/firefox-trunk.head/view/head:/debian/patches/unity-menubar.patch

**Comment by criztovyl** [comment]
Also, can you post the specific linker error? :)

**Comment by PandaCodex** [comment]
**[Latest stable release version version](http://bazaar.launchpad.net/~mozillateam/firefox/firefox.trusty/view/head:/debian/patches/unity-menubar.patch)**

**[Build revision 56.0](http://bazaar.launchpad.net/~mozillateam/firefox/firefox.trusty/view/1200/debian/patches/unity-menubar.patch)**

**Try these patches, same ones [Cyberfox](https://github.com/InternalError503/cyberfox/blob/master/_Build/_Linux/unity/README-Unity-Menu-Bar-Patch.txt) uses but ones for the latest stable release `57.0` branch the other is targeted for `56.0` branch same one as Waterfox.**

**Comment by darthoctopus** [comment]
```
../../widget/gtk/nsMenu.o: In function `nsMenu::OnAttributeChanged(nsIContent*, nsIAtom*)':
<redacted>/Waterfox/widget/gtk/nsMenu.cpp:(.text._ZN6nsMenu18OnAttributeChangedEP10nsIContentP7nsIAtom+0x124): undefined reference to `nsStyleContext::Release()'
/usr/bin/ld: libxul.so: hidden symbol `_ZN14nsStyleContext7ReleaseEv' isn't defined
```

I've noticed that the patch I'm using differs slightly from the 56.0 one that @PandaCodex linked; will try that and report shortly

**Comment by darthoctopus** [comment]
update: it works, I'm an idiot D:

For future reference: I was using revision 1189, which was missing an include for "nsStyleContextInlines.h" (this is probably the reason for the linker error).

However, I would like to keep open the possibility of merging in the global menu by default.

**Comment by hawkeye116477** [comment]
It can be closed, cuz now Global Menu integration is available since 56.2.10.1.
https://github.com/MrAlex94/Waterfox/commit/d7201d404758aa6222f80502e3dcdf016fc87a31

**Comment by darthoctopus** [comment]
> It can be closed, cuz now Global Menu integration is available since 56.2.10.1.
> [d7201d4](https://github.com/MrAlex94/Waterfox/commit/d7201d404758aa6222f80502e3dcdf016fc87a31)

thank you! I have verified this to work on the version of waterfox currently available for download..
